### PR TITLE
fix: Inbox name validation with regex format for special characters

### PR DESCRIPTION
**Pull request by tejaswinichile**

### CODE REVIEW

File: app/models/inbox.rb [hunk 1]
```diff
@@ -33,7 +33,10 @@ class Inbox < ApplicationRecord
   include Avatarable
   include OutOfOffisable
 
-  validates :name, presence: true, format: { with: /^\b[\w\s]*\b$/, multiline: true }
+  # Not allowing characters:
+  validates :name, presence: true,
+                   format: { with: %r{^^\b[^/\\<>@]*\b$}, multiline: true,
+                             message: 'Should not start and end with symbols and should not have < > / \ @ symbols' }
```
Comment (pranavrajs) [inline app/models/inbox.rb:39]:
```suggestion
                             message: 'The name should not start or end with symbols, and it should not have < > / \ @ characters.' }
```

Comment (tejaswinichile) [reply app/models/inbox.rb:39]: @pranavrajs Name is auto included in the validation message, as it is the attribute name.

**Comment by pranavrajs** [comment]
@tejaswinichile It looks like FB/Twitter allows some of these characters. We should scope the validation to email, website, and API channels only.

image

File: spec/models/inbox_spec.rb [hunk 2]
```diff
@@ -164,13 +165,31 @@
...
 
         inbox.name = 'inbox_name.and_1'
         expect(inbox).to be_valid
       end
+
+      context 'when special characters allowed for some channel' do
+        let!(:tw_channel_val) { FactoryBot.create(:channel_twitter_profile) }
+        let(:inbox) { create(:inbox, channel: tw_channel_val) }
+
+        it 'does allow chacters like /\@<> for Facebook Channel' do
```
Comment (pranavrajs) [inline spec/models/inbox_spec.rb:188]:
```suggestion
        it 'does allow special characters like /\@<> for Facebook Channel' do
```

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**